src/routes/authStrategies/LocalStrategy.ts

```diff
@@ -0,0 +1,48 @@
+import db from "../../../db/db.connection";
+import LocalStrategy from "passport-local";
+import bcrypt from "bcrypt";
+
+const local = new LocalStrategy.Strategy(
+  {
+    usernameField: "email",
+  },
+  async (username, password, done) => {
+    try {
+      const [selectedUser] = await db("user").select().where("email", username);
+
+      if (!selectedUser) {
+        await bcrypt.genSalt(10, async (err, salt) => {
+          if (err) {
+            console.log(err);
+          } else {
+            await bcrypt.hash(password, salt, async (err, hash) => {
+              if (err) {
+                console.log(err);
+              } else {
+                const [newUser] = await db("user").returning(["*"]).insert({
```

Comment:
Can't be sending the whole user object through req.user as that will contain the password!